test/unit/BCoWPool.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {BasePoolTest} from './BPool.t.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
+import {MockBCoWPool} from 'test/manual-smock/MockBCoWPool.sol';
+import {MockBPool} from 'test/smock/MockBPool.sol';
+
+import {Test} from 'forge-std/Test.sol';
+
+abstract contract BaseCoWPoolTest is BasePoolTest {
+  address public cowSolutionSettler = makeAddr('cowSolutionSettler');
+  bytes32 public domainSeparator = bytes32(bytes2(0xf00b));
+  address public vaultRelayer = makeAddr('vaultRelayer');
+
+  function setUp() public override {
+    super.setUp();
+    vm.mockCall(cowSolutionSettler, abi.encodePacked(ISettlement.domainSeparator.selector), abi.encode(domainSeparator));
+    vm.mockCall(cowSolutionSettler, abi.encodePacked(ISettlement.vaultRelayer.selector), abi.encode(vaultRelayer));
+    bPool = MockBPool(address(new MockBCoWPool(cowSolutionSettler)));
+  }
+}
+
+contract BCoWPool_Unit_Constructor is BaseCoWPoolTest {
+  function test_Set_SolutionSettler(address _settler) public {
+    assumeNotForgeAddress(_settler);
+    vm.mockCall(_settler, abi.encodePacked(ISettlement.domainSeparator.selector), abi.encode(domainSeparator));
+    vm.mockCall(_settler, abi.encodePacked(ISettlement.vaultRelayer.selector), abi.encode(vaultRelayer));
```

Comment:
why not on the `setup()`?

Reply:
the fuzz test asserts the return value from the `domainSeparator` and `vaultRelayer` methods on the `_settler` are called. If I mock the calls on the `setUp` then I can't fuzz the `_settler` address and tests would pass if the address was also hard-coded in the implementation.

It's very minor, yes, but does cover an intended behaviour